File: iosApp/iosApp/Presentation/Cafe/Event/CafeEventUiState.swift
//
//  CafeEventUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 4/23/26.
//

import Foundation
import Shared

struct CafeEventUiState {
    var isLoading: Bool = false
    var event: CafeEventManagementItem? = nil
    var errorMessage: String? = nil
    var participantCasts: [Cast] = []
    var isLikedByMe: Bool = false
    var likeCount: Int = 0
    var isTogglingLike: Bool = false

    static let empty = CafeEventUiState()
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/cafe/event/CafeEventUiState.kt
package com.hhp227.concafe.presentation.cafe.event

import com.hhp227.concafe.domain.model.Cast
import com.hhp227.concafe.domain.model.CafeEventManagementItem

data class CafeEventUiState(
    val isLoading: Boolean = false,
    val event: CafeEventManagementItem? = null,
    val errorMessage: String? = null,
    val participantCasts: List<Cast> = emptyList(),
    val isLikedByMe: Boolean = false,
    val likeCount: Int = 0,
    val isTogglingLike: Boolean = false,
    val isLoggedIn: Boolean = false,
    val isLoginPromptVisible: Boolean = false
) {
    companion object {
        fun empty() = CafeEventUiState()
    }
}
